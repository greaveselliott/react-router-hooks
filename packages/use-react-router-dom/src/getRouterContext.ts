import { __RouterContext } from "react-router";

const getRouterContext = () => {
  if (!__RouterContext) {
    throw new Error(
      "useReactRouter may only be called within a <Router /> context."
    );
  }

  return __RouterContext;
};

export default getRouterContext;
