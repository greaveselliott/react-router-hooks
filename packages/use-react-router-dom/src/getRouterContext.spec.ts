import { __RouterContext } from "react-router";
import getRouterContext from "./getRouterContext";

jest.mock("react-router", () => ({
  __RouterContext: undefined
}));

test("Throw an error if this method is not called within a <Router /> context.", () => {
  expect(() => getRouterContext()).toThrowErrorMatchingSnapshot();
});
