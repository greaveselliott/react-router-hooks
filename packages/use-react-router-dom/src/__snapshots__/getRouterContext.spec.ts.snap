// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Throw an error if this method is not called within a <Router /> context. 1`] = `"useReactRouter may only be called within a <Router /> context."`;
